### PR TITLE
Optimizespecs

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

**Comment by alissapajer** [comment]
@sellout this PR contains #1354 - just the HEAD commit is new, but it's a separate change, hence the new PR.

File: core/src/test/scala/quasar/qscript/Optimize.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    "optimize a basic read" in {
+      import QueryFile.optimize._
+
+      val run =
+        (quasar.fp.free.injectedNT[QS](simplifyProjections).apply(_: QS[Fix[QS]])) ⋙
+          liftFG(coalesceMapShift[QS, QS](optionIdF[QS])) ⋙
+          Normalizable[QS].normalize ⋙
+          liftFG(simplifySP[QS, QS](optionIdF[QS])) ⋙
+          liftFG(compactLeftShift[QS, QS])
```
Comment (alissapajer) [inline core/src/test/scala/quasar/qscript/Optimize.scala]: In a test like this with multiple optimizations being run, there is no guarantee that all the optimizations have an effect (i.e. are necessary). Hence I'm also working on tests that each just run a single optimization, to be clear what each optimization does (and doesn't) do.

**Comment by sellout** [comment]
This looks good. I don’t want to merge it until #1354 is ready for merge, though.

**Comment by alissapajer** [comment]
Hm I thought when #1354 merged, the diff would update here to only show the commits unique to this PR...

**Comment by sellout** [comment]
I think this just needs master merged into it again? Not that that’ll fix the GitHub diff, but it’ll allow me to merge.

**Comment by alissapajer** [comment]
@sellout that was it - I merged master in and now it just show the commits unique to this PR!

**Pull request merged**